{
  "info": {
    "author": "kerol",
    "author_email": "ikerol@163.com",
    "bugtrack_url": null,
    "classifiers": [
      "License :: OSI Approved :: MIT License",
      "Operating System :: OS Independent",
      "Programming Language :: Python :: 3"
    ],
    "description": "### Django-conn-pool\nMySQL Connection Pooling with Django(<2.0.0) and SQLAlchemy(>=1.2.7).\n\n### Why\nIf `CONN_MAX_AGE` not set in you settings, Django will establish a new MySQL connection for each request and close it after the request.\n\nSo if where are too many connections to be handled, it will establish too many connections until running out of MySQL `max_connections`.\n\nThen, MySQL will return the error: `ERROR 1040 (08004): Too many connections`.\n\nIt's necessary to maintain a connection pool for a robust Django project.\n\n### Principle\n[Reference: MySQL Connection Pooling with Django and SQLAlchemy](http://menendez.com/blog/mysql-connection-pooling-django-and-sqlalchemy/)\n\nMake MySQLdb Database `get_new_connection` from connection pool with SQLAlchemy QueuePool.\n\nSo all we need to custom a Database managed by connection pool and get new connection from the pool.\n\n```\npip install mysqlclient\npip install SQLAlchemy\n```\n\n### Usage\n\n```\npip install django-conn-pool\n```\n\nsettings.py\n\n```\n# http://docs.sqlalchemy.org/en/latest/core/pooling.html#sqlalchemy.pool.QueuePool\n# http://docs.sqlalchemy.org/en/latest/core/pooling.html#sqlalchemy.pool.Pool.params\nSQLALCHEMY_QUEUEPOOL = {\n    'pool_size': 10,\n    'max_overflow': 10,\n    'timeout': 5,\n    'recycle': 119,\n}\n\nDATABASES = {\n    'default': {\n        #'ENGINE': 'django.db.backends.mysql',\n        'ENGINE': 'django_conn_pool.mysql',\n        'HOST': '127.0.0.1',\n        'NAME': 'xxx',\n        'USER': 'xxx',\n        'PASSWORD': 'xxx',\n        'PORT': 3306,\n    },\n}\n```\n\n\n### Test\nIf `python manage.py runserver` is OK, now you can use load testing utility [Siege](https://www.joedog.org/siege-home/) to test.\n\n```\nsiege -r 2 -c 1000 -d 0 http://xxx  # siege --help\n```\n\nYou can compare the MySQL process list count to the before, and it should work.\n\nBtw, you should get a `sqlalchemy.exc.TimeoutError:` exception if it does not get a connection during the request.\n\n\n",
    "description_content_type": "text/markdown",
    "docs_url": null,
    "download_url": "",
    "downloads": {
      "last_day": -1,
      "last_month": -1,
      "last_week": -1
    },
    "home_page": "https://github.com/kerol/django-conn-pool",
    "keywords": "",
    "license": "",
    "maintainer": "",
    "maintainer_email": "",
    "name": "django-conn-pool",
    "package_url": "https://pypi.org/project/django-conn-pool/",
    "platform": "",
    "project_url": "https://pypi.org/project/django-conn-pool/",
    "project_urls": {
      "Homepage": "https://github.com/kerol/django-conn-pool"
    },
    "release_url": "https://pypi.org/project/django-conn-pool/1.11.1/",
    "requires_dist": [
      "Django (<2.0.0)",
      "SQLAlchemy (>=1.2.7)"
    ],
    "requires_python": "",
    "summary": "Connection Pooling with Django and SQLAlchemy",
    "version": "1.11.1",
    "yanked": false,
    "yanked_reason": null
  },
  "last_serial": 6331030,
  "urls": [
    {
      "comment_text": "",
      "digests": {
        "blake2b_256": "d02cb6fe0485a6e52ed1c709f4f59ea644cc5e14706293c949c9c148f66b627a",
        "md5": "3fc0eaf3e0232ceb08cac88642460c83",
        "sha256": "196bfde83f829bf20894fe2ef5a2ab8404be250727db5dcef6affb497c50f2fb"
      },
      "downloads": -1,
      "filename": "django_conn_pool-1.11.1-py3-none-any.whl",
      "has_sig": false,
      "md5_digest": "3fc0eaf3e0232ceb08cac88642460c83",
      "packagetype": "bdist_wheel",
      "python_version": "py3",
      "requires_python": null,
      "size": 5236,
      "upload_time": "2018-05-29T12:42:53",
      "upload_time_iso_8601": "2018-05-29T12:42:53.862444Z",
      "url": "https://files.pythonhosted.org/packages/d0/2c/b6fe0485a6e52ed1c709f4f59ea644cc5e14706293c949c9c148f66b627a/django_conn_pool-1.11.1-py3-none-any.whl",
      "yanked": false,
      "yanked_reason": null
    },
    {
      "comment_text": "",
      "digests": {
        "blake2b_256": "fa8ecec86a3f53551de82a2f58ee8e8782c07eef2ac86afb7d899f4334e9a8ce",
        "md5": "4ed5221e713ed66283cf3b8be70e28d4",
        "sha256": "8c4c3bfc870da2cdd960c0f02913d3e07ebf913f3acc25ea5c9e61ab57bdf8e8"
      },
      "downloads": -1,
      "filename": "django-conn-pool-1.11.1.tar.gz",
      "has_sig": false,
      "md5_digest": "4ed5221e713ed66283cf3b8be70e28d4",
      "packagetype": "sdist",
      "python_version": "source",
      "requires_python": null,
      "size": 4326,
      "upload_time": "2018-05-29T12:42:55",
      "upload_time_iso_8601": "2018-05-29T12:42:55.572143Z",
      "url": "https://files.pythonhosted.org/packages/fa/8e/cec86a3f53551de82a2f58ee8e8782c07eef2ac86afb7d899f4334e9a8ce/django-conn-pool-1.11.1.tar.gz",
      "yanked": false,
      "yanked_reason": null
    }
  ],
  "vulnerabilities": []
}